internal/store/http_publish.go

```
@@ -0,0 +1,122 @@
+	if int64(len(tarball)) > maxArtifactBytes {
+		return fmt.Errorf("artifact for %s exceeds %d bytes", b.Name, int64(maxArtifactBytes))
+	}
+	sum := sha256hex(tarball)
+
+	kind := p.Kind
+	if kind == "" {
+		kind = "button"
+	}
+	endpoint := strings.TrimRight(p.BaseURL, "/") +
+		fmt.Sprintf("/v1/buttons/%s/%s", scopedPath(b.Name), url.PathEscape(b.Version))
+	req, err := http.NewRequest(http.MethodPost, endpoint, bytes.NewReader(tarball))
```

golangci-lint (2.12.2)

[error] net/http.NewRequest must not be called. use net/http.NewRequestWithContext

(noctx)